src/serialization/v2_serializer.rs

```diff
@@ -14,15 +14,32 @@ pub enum V2SerializeError {
     /// hardware.
     UsizeTypeTooSmall,
     /// An i/o operation failed.
-    IoError(ErrorKind),
+    IoError(io::Error),
 }
 
 impl std::convert::From<std::io::Error> for V2SerializeError {
     fn from(e: std::io::Error) -> Self {
-        V2SerializeError::IoError(e.kind())
+        V2SerializeError::IoError(e)
     }
 }
 
+impl fmt::Display for V2SerializeError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            V2SerializeError::CountNotSerializable => write!(
+                f,
+                "A count above i64::max_value() cannot be zig-zag encoded"
+            ),
+            V2SerializeError::UsizeTypeTooSmall => {
+                write!(f, "Internal calculations cannot be represented in `usize`")
+            }
+            V2SerializeError::IoError(e) => write!(f, "An i/o operation failed: {}", e),
+        }
+    }
+}
+
+impl error::Error for V2SerializeError {}
```

Comment:
This should override the implementation of [`Error::source`](https://doc.rust-lang.org/std/error/trait.Error.html#method.source).